e2e/test/scenarios/question-reproductions/reproductions-3.cy.spec.js

```diff
@@ -1737,6 +1739,238 @@ describe("issue 39771", () => {
+      type: "external",
+      name: remappedDisplayName,
+      human_readable_field_id: targetFieldId,
+    });
+  }
+
+  function verifyListFilter({ fieldDisplayName, fieldValue, fieldValueLabel }) {
+    tableHeaderClick(fieldDisplayName);
+    popover().findByText("Filter by this column").click();
+    popover().within(() => {
+      cy.findByPlaceholderText("Search the list").type(fieldValueLabel);
+      cy.findByText(fieldValueLabel).click();
+      cy.button("Add filter").click();
+    });
+    cy.findByTestId("qb-filters-panel")
+      .findByText(`${fieldDisplayName} is ${fieldValue}`)
+      .click();
+    popover().findByLabelText(fieldValueLabel).should("be.checked");
+  }
+
+  function verifySearchFilter({
+    fieldDisplayName,
+    fieldValue,
+    fieldValueLabel,
+  }) {
+    tableHeaderClick(fieldDisplayName);
+    popover().findByText("Filter by this column").click();
+    popover()
+      .findByPlaceholderText(`Search by ${fieldDisplayName}`)
+      .type(fieldValueLabel);
+    popover().last().findByText(fieldValueLabel).click();
+    popover().first().click().button("Add filter").click();
+    cy.findByTestId("qb-filters-panel")
+      .findByText(`${fieldDisplayName} is ${fieldValue}`)
+      .click();
+    popover().findByText(fieldValueLabel).should("be.visible");
+  }
+
+  function verifyRemappedFilter({
+    visitCard,
+    fieldId,
+    fieldDisplayName,
+    fieldValue,
+    fieldValueLabel,
+    expectedRowCount,
+  }) {
+    cy.log("list values");
+    cy.signInAsAdmin();
+    setListValues({ fieldId });
+    cy.signInAsNormalUser();
+    visitCard();
+    verifyListFilter({ fieldDisplayName, fieldValue, fieldValueLabel });
+    assertQueryBuilderRowCount(expectedRowCount);
+
+    cy.log("search values");
+    cy.signInAsAdmin();
+    setSearchValues({ fieldId });
+    cy.signInAsNormalUser();
+    visitCard();
+    verifySearchFilter({ fieldDisplayName, fieldValue, fieldValueLabel });
+    assertQueryBuilderRowCount(expectedRowCount);
+  }
+
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+  });
+
+  describe("type/PK -> type/Name remapping (metabase#45063)", () => {
+    it("should work with questions", () => {
+      createGuiQuestion({ sourceTableId: PEOPLE_ID });
+      verifyRemappedFilter({
+        visitCard: () => visitQuestion("@questionId"),
+        fieldId: PEOPLE.ID,
+        fieldDisplayName: "ID",
+        fieldValue: 1,
+        fieldValueLabel: "Hudson Borer",
+        expectedRowCount: 1,
+      });
+    });
+
+    it("should work with models", () => {
+      createGuiModel({ sourceTableId: PEOPLE_ID });
+      verifyRemappedFilter({
+        visitCard: () => cy.get("@modelId").then(visitModel),
+        fieldId: PEOPLE.ID,
+        fieldDisplayName: "ID",
+        fieldValue: 1,
+        fieldValueLabel: "Hudson Borer",
+        expectedRowCount: 1,
+      });
+    });
+
+    it("should work with native models", () => {
```

Comment:
Should we assert native questions as well?

Reply:
different component / bug